test/compilable/b19294.d

```diff
@@ -1,69 +1,180 @@
-alias MT = MyStruct!int;
+alias T = MyStruct!float;
 
-struct MyStruct(T)
+struct MyStruct(U)
 {
-    T x;
+    U x;
+    U y;
 
-    this(T y)
+    this(U xx, U yy)
     {
-        x = y;
+        x = xx;
+        y = yy;
     }
 
-    MyStruct!T opBinary(string op)(MyStruct!T y) const
+    MyStruct!U opBinary(string op)(MyStruct!U z) const
     {
         alias C = typeof(return);
-        auto w = C(this.x);
-        return w.opOpAssign!(op)(y);
+        auto w = C(this.x, this.y);
+        return w.opOpAssign!(op)(z);
     }
 
-    MyStruct!T opBinaryRight(string op)(MyStruct!T y) const
+    MyStruct!U opBinaryRight(string op)(MyStruct!U z) const
     {
-        return opBinary!(op)(y);
+        return opBinary!(op)(z);
     }
 
-    ref MyStruct opOpAssign(string op, T)(const MyStruct!T z)
+    ref MyStruct opOpAssign(string op, U)(const MyStruct!U z)
     {
         mixin ("x "~op~"= z.x;");
+        mixin ("y "~op~"= z.y;");
         return this;
     }
 
-    MyStruct!T opBinary(string op)(T y) const
+    MyStruct!U opBinary(string op, R)(R z) const
+        if (is(R == int) || is(R == float))
     {
         alias C = typeof(return);
-        auto w = C(this.x);
-        return w.opOpAssign!(op)(y);
+        auto w = C(this.x, this.y);
+        return w.opOpAssign!(op)(z);
     }
 
-    MyStruct!T opBinaryRight(string op)(T y) const
+    MyStruct!U opBinaryRight(string op, R)(R z) const
+        if (is(R == int) || is(R == float))
     {
-        return opBinary!(op)(y);
+        return opBinary!(op)(z);
     }
 
-    ref MyStruct opOpAssign(string op, T)(const T z)
+    ref MyStruct opOpAssign(string op, R)(const R z)
+        if (is(R == int) || is(R == float))
     {
         mixin ("x "~op~"= z;");
         return this;
     }
 }
 
-void test()
+void main()
```

Comment:
if this is supposed to be a runnable test move the file to runnable, otherwise leave the name unchanged.